File: codigo/backend/go-backend/models/produtocarrinho.go
package models

import "fmt"


type ProdutoCarrinho struct {
    CarrinhoID uint   `gorm:"primaryKey"`
    ProdutoID  uint   `gorm:"primaryKey"`
    Quantidade int    `gorm:"not null"`
    Produto    Produto `gorm:"foreignKey:ProdutoID"`
}


func (ProdutoCarrinho) TableName() string {
    return "produtoscarrinhos"
}

func (pc *ProdutoCarrinho) ToMap() map[string]interface{} {
    return map[string]interface{}{
        "carrinho_id": pc.CarrinhoID,
        "produto_id":  pc.ProdutoID,
        "quantidade":  pc.Quantidade,
    }
}

func (pc *ProdutoCarrinho) String() string {
    return "<ProdutoCarrinho " + fmt.Sprintf("%d-%d", pc.CarrinhoID, pc.ProdutoID) + ">"
}

File: codigo/backend/sap-ecc-emulator/src/models/produtocarrinho.py
from db import db

class ProdutoCarrinho(db.Model):
    __tablename__ = 'produtocarrinho'

    carrinho_id = db.Column(db.Integer, primary_key=True)
    produto_id = db.Column(db.Integer, db.ForeignKey('produtos.id'), primary_key=True)
    quantidade = db.Column(db.Integer, nullable=False)

    produto = db.relationship('Produto', backref='produtocarrinho', lazy=True)

    def to_dict(self):
        return {
            'carrinho_id': self.carrinho_id,
            'produto_id': self.produto_id,
            'quantidade': self.quantidade
        }

    def __repr__(self):
        return f'<ProdutoCarrinho {self.carrinho_id}-{self.produto_id}>'
